### PR TITLE
Release 2.0.0-rc4

### PR DESCRIPTION
**Pull request by mkindahl**
This release candidate contains bugfixes since the previous release
candidate, as well as additional minor features. It improves
validation of configuration changes for background jobs, adds support
for gapfill on distributed tables, contains improvements to the memory
handling for large COPY, and contains improvements to compression for
distributed hypertables.

**Minor Features**
* #2689 Check configuration in alter_job and add_job
* #2696 Support gapfill on distributed hypertable
* #2468 Show more information in get_git_commit
* #2678 Include user actions into job stats view
* #2664 Fix support for complex aggregate expression
* #2672 Add hypertable to continuous aggregates view
* #2662 Save compression metadata settings on access node
* #2707 Introduce additional db for data node bootstrapping

**Bugfixes**
* #2688 Fix crash for concurrent drop and compress chunk
* #2666 Fix timeout handling in async library
* #2683 Fix crash in add_job when given NULL interval
* #2698 Improve memory handling for remote COPY
* #2555 Set metadata for chunks compressed before 2.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,26 @@
 `psql` with the `-X` flag to prevent any `.psqlrc` commands from
 accidentally triggering the load of a previous DB version.**
 
+## 2.0.0-rc4 (2020-12-02)
+
+This release candidate contains bugfixes since the previous release
```
Comment (erimatnor) [inline CHANGELOG.md:9]: I would expand (a bit dry right now). Maybe say something about improvements to COPY, support for gapfill, and improvements to chunk compression on distributed hypertables?

**Review by erimatnor — COMMENTED** [review]
The new update script is missing.

File: version.config [hunk 2]
```diff
@@ -1,2 +1,3 @@
-version = 2.1.0-dev
+version = 2.0.0-rc4
```
Comment (erimatnor) [inline version.config:1]: This should not change here.

**Comment by mkindahl** [comment]
> The new update script is missing.

Added.

File: CHANGELOG.md (hunk 1 above)
Comment (mkindahl) [reply CHANGELOG.md:9]: Rewrote it. I'll add info about COPY when it is merged.

File: version.config (hunk 2 above)
Comment (mkindahl) [reply version.config]: Fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,28 @@
...
+candidate, as well as additional minor features. It adds checks of
```
Comment (erimatnor) [inline CHANGELOG.md:10]: unclear what "checks of configuration" is or refers to. Maybe "It improves configuration validation for background jobs, ..."

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,28 @@
...
+* #2664 Fix support for complex aggregate expression
+* #2672 Add hypertable to continuous aggregates view
+
+**Bugfixes**
+* #2688 Fix crash for concurrent drop and compress chunk
+* #2666 Fix timeout handling in async library
+* #2683 Fix crash in add_job when given NULL interval
+
+**Thanks**
+
```
Comment (erimatnor) [inline CHANGELOG.md:28]: Should add acknowledgements.

File: CHANGELOG.md (hunk 3 above)
Comment (mkindahl) [reply CHANGELOG.md]: Reformulated.

File: CHANGELOG.md (hunk 4 above)
Comment (mkindahl) [reply CHANGELOG.md:36]: Added two found in associated issues.

**Review by erimatnor — APPROVED** [review]

**Review by pmwkaa — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,37 @@
...
+* #2688 Fix crash for concurrent drop and compress chunk
+* #2666 Fix timeout handling in async library
+* #2683 Fix crash in add_job when given NULL interval
+* #2698 Improve memory handling for remote COPY
+* #2555 Set metadata for chunks compressed before 2.0
+
+**Thanks**
+* @francesco11112 for reporting memory issue on COPY
+* @Netskeh for reporting bug on time_bucket problem in continuous
+  aggregates
```
Comment (k-rus) [inline CHANGELOG.md:36]: There was no problem with `time_bucket` itself but with using it in an expression.
```suggestion
* @Netskeh for reporting a bug with expressions over time_bucket in continuous
   aggregates
```

**Review by k-rus — COMMENTED** [review]
I suggest to add that bug #2616 was fixed

**Pull request merged**